### PR TITLE
Add `check_similarity`

### PR DESCRIPTION
**Pull request by koviubi56**
This function returns the similarity of two strings.

**Note**: I'm assuming that with this function you want to verify whether the user's input is correct. If this is the case, then it's probably not the best idea.

### CODE REVIEW

File: src/sincere_singularities/utils.py [hunk 1]
```diff
@@ -51,3 +52,17 @@ def load_json(filename: str, json_type: type[T]) -> T:
         typed_json: T = dacite.from_dict(json_type, loaded_json)
 
         return typed_json
+
+
+def check_similarity(first: str, second: str) -> float:
+    """
+    Measure of the strings' similarity as a float.
+
+    Args:
```
Comment (WassCodeur) [inline src/sincere_singularities/utils.py:73]: when you annotate the function, do you still have to describe the parameters? I find it a bit redundant.

**Review by WassCodeur — APPROVED** [review]

File: src/sincere_singularities/utils.py (hunk 1 above)
Comment (koviubi56) [reply src/sincere_singularities/utils.py:73]: Yes.

Comment (WassCodeur) [reply src/sincere_singularities/utils.py:73]: OKay!

just that it's a bit of a rehearsal for me

File: src/sincere_singularities/utils.py [hunk 2]
```diff
@@ -51,3 +52,17 @@ def load_json(filename: str, json_type: type[T]) -> T:
...
+        first (str): The first string.
+        second (str): The second string.
+
+    Returns:
+        float: The similarity of the two strings [0, 1]
+    """
+    return difflib.SequenceMatcher(None, first, second).ratio()
```
Comment (Vinyzu) [inline src/sincere_singularities/utils.py:80]: Sequence Matcher wont work.
It uses the [Gestalt Pattern Matching Algorithm](https://en.wikipedia.org/wiki/Gestalt_pattern_matching).
So it doesnt actually use any Sementic Analysis/Sentence-Matching AI, but just math.

For example: with these two sentences:
```py
>>> str0 = "The cat is on the mat."
>>> str1 = "A feline rests atop a rug."
```
(These sentences are very unrelated to the project, but show the point. Sentences generated with ChatGPT.)

I get a SequenceMatcher result of:
```py
>>> difflib.SequenceMatcher(None, str1, str0).ratio()
0.2127659574468085
```

But for example [Huggingface Sentence-Similarity Demo](https://huggingface.co/tasks/sentence-similarity) gives me 0.57, which is much closer.
So we have to use a (sentence-transformer AI for this...)

**Review by Vinyzu — CHANGES_REQUESTED** [review]
Implement a (sentence-transformer) AI.

File: src/sincere_singularities/utils.py (hunk 2 above)
Comment (koviubi56) [reply src/sincere_singularities/utils.py:80]: I'll leave that up to you.

**Review by Vinyzu — APPROVED** [review]
LGTM!

**Pull request merged**